app/code/Magento/QuoteGraphQl/Model/Cart/AddSimpleProductToCart.php

```diff
@@ -126,20 +125,51 @@ private function extractQty(array $cartItemData): float
     /**
      * Extract Customizable Options from cart item data
      *
+     * @param \Magento\Catalog\Model\Product $product
      * @param array $cartItemData
      * @return array
+     * @throws GraphQlInputException
      */
-    private function extractCustomizableOptions(array $cartItemData): array
+    private function extractCustomizableOptions(\Magento\Catalog\Model\Product $product, array $cartItemData): array
     {
         $customizableOptions = $this->arrayManager->get('customizable_options', $cartItemData, []);
-
+        $optionTypeDetails = $this->getOptionTypeDetails($product);
         $customizableOptionsData = [];
-        foreach ($customizableOptions as $customizableOption) {
-            $customizableOptionsData[$customizableOption['id']] = $customizableOption['value'];
+
+        foreach ($customizableOptions as $optionItem) {
+            if (!isset($optionItem['value'])) {
+                throw new GraphQlInputException(__('Value is required for Option Id %1', $optionItem['id']));
+            }
+
+            $optId = $optionItem['id'];
+            if (in_array($optionTypeDetails[$optId], ["checkbox", "multiple"])) {
+                $customizableOptionsData[$optionItem['id']][] = $optionItem['value'];
+                //Sort needed to prevent add option twice with same options different ways
+                //Eg. {id: 1, value: "1"},{id: 1, value: "2"} and {id: 1, value: "2"},{id: 1, value: "1"}
+                arsort($customizableOptionsData[$optionItem['id']]);
```

Comment:
Why do we need to sort input data?

Reply:
Otherwise below 2 options create 2 cart items

customizable_options: [
                                      {id: 1, value: "1"},
                                      {id: 1, value: "2"}
                                    ]

customizable_options: [
                                      {id: 1, value: "2"},
                                      {id: 1, value: "1"}
                                    ]